pages/shopping/index.js

```diff
@@ -80,26 +112,61 @@ export default function ShoppingList({ user, mutateUser }) {
           </StyledListItem>
         ) : (
           user.shoppingList.map((item, index) => (
-            <StyledListItem key={index}>
-              <StyledCheck>
-                <StyledNumberUnit>
-                  <StyledCheckItem $text={item.isChecked} $flex={0.1}>
-                    {item.quantity}
-                  </StyledCheckItem>
-                  <StyledCheckItem $text={item.isChecked} $flex={1}>
-                    {item.unit}
-                  </StyledCheckItem>
-                </StyledNumberUnit>
-                <StyledCheckItem $text={item.isChecked} $flex={2}>
-                  {item.name}
-                </StyledCheckItem>
-              </StyledCheck>
-
-              <StyledCheckbox
-                type="checkbox"
-                checked={item.isChecked}
-                onChange={() => handleCheckboxChange(index)}
-              ></StyledCheckbox>
+            <StyledListItem key={index} onClick={() => handleItemClick(index)}>
+              {editingIndex === index ? (
+                <StyledEditForm
+                  ref={editFormRef}
+                  onSubmit={(event) => handleItemEdit(event, index)}
+                >
+                  <StyledInput
+                    type="number"
+                    defaultValue={item.quantity}
+                    min="0"
+                    aria-label="edit ingredient quantity for the recipe"
+                    name="quantity"
+                  />
+                  <StyledDropDown name="unit" defaultValue={item.unit}>
+                    <option value="">-</option>
+                    <option value="ml">ml</option>
+                    <option value="piece">Stück</option>
+                    <option value="gramm">g</option>
+                    <option value="EL">EL</option>
+                    <option value="TL">TL</option>
+                    <option value="Prise">Prise</option>
+                  </StyledDropDown>
```

Comment:
This reduces repetition and makes it easier to add or remove units in the future. 

```js
// Define your units in an array outside the component for cleanliness
const unitOptions = ["ml", "piece", "gramm", "EL", "TL", "Prise"];

// ... other code

                  <StyledDropDown name="unit" defaultValue={item.unit}>

                  <option value="">-</option>
                  {unitOptions.map((unit) => (
                    <option key={unit} value={unit}>
                      {unit}
                    </option>  // Mapping over unitOptions to create option elements
                  ))}
                </StyledDropDown>


// ... other code
```

Reply:
Thank you Jessica for your very detailed review and the explanations. I implemented your suggestions right away!:) 
I definitely agree to your point in advising us to focus on labels for our input fields. For accessibility matters we use here aria-label. My question is: is this procedure not sufficient? Implementing labels is difficult because there is no space for them in the frontend of the shopping list.